esp-hal-common/src/analog/adc/riscv.rs

```diff
@@ -646,6 +646,19 @@ where
             let attenuation = self.attenuations[channel as usize].unwrap() as u8;
             ADCI::config_onetime_sample(channel, attenuation);
             ADCI::start_onetime_sample();
+
+            // see https://github.com/espressif/esp-idf/blob/b4268c874a4cf8fcf7c0c4153cffb76ad2ddda4e/components/hal/adc_oneshot_hal.c#L105-L107
+            // the delay might be a bit generous but longer delay seem to not cause problems
+            #[cfg(esp32c6)]
+            {
+                extern "C" {
+                    fn ets_delay_us(us: u32);
+                }
+                unsafe {
+                    ets_delay_us(40);
+                }
+                ADCI::start_onetime_sample();
```

Comment:
This isn't the same sequence as the linked C code: that is a `reg = false; wait; reg = true` sequence, while this looks like two bit sets. It's interesting how the workaround still seems to work, I wonder what actually needs the delay.

Reply:
Yes, it's not the same thing - but works